src/security/Io.cc

```diff
@@ -83,10 +83,15 @@ Security::Handshake(Comm::Connection &transport, const ErrorCode topError, Fun i
         ; // fall through to handle the problem
     }
 
-    // now we know that we are dealing with a real problem; detail it
-    const ErrorDetail::Pointer errorDetail =
-        new ErrorDetail(topError, ioError, xerrno);
-
+    ErrorDetail::Pointer errorDetail;
+    if (const auto verifyErrorDetailRaw = SSL_get_ex_data(connection, ssl_ex_index_ssl_error_detail)) {
+        errorDetail = *static_cast<ErrorDetail::Pointer*>(verifyErrorDetailRaw);
+    } else {
+        // now we know that we are dealing with a real problem; detail it
```

Comment:
This comment looks out of place after the move because that `now` was true outside/before this `if` statement. Let's move it before the `if` statement, where it used to be. I think it is OK that we are reusing old details in some cases. We are still detailing the problem we just found (by providing those existing details).

Reply:
commit c4de550fe3